kie-api/src/main/java/org/kie/api/builder/KieBuilder.java

```diff
@@ -39,6 +41,13 @@
     KieBuilder buildAll();
 
     /**
+     * Builds all the KieBases contained in the KieModule for which this KieBuilder has been created
+     * @param classFilters Used to prevent compilation of Java source files.
+     *          A Java source file vetoed by any of the given filters will not be compiled.
+     */
+    KieBuilder buildAll(Collection<KieClassFilter> classFilters);
```

Comment:
@mbarkley @mswiderski I'm not understanding why you need a Collection of filters here. Isn't a single one enough? Also I'm assuming that this is only for Drools 7 and then we are allowed to use Java 8 features since you used a Stream in the KieBuilder implementation. If so this filter is actually just a Predicate, so my suggestion is to get rid of the new KieClassFilter interface, use the following signature for this method

 `KieBuilder buildAll(Predicate<String> classFilters);`

and change the implementation accordingly.